### PR TITLE
Remove smoke E2E test suite

### PR DESCRIPTION
**Pull request by iblancasa**
Signed-off-by: Israel Blancas <iblancas@redhat.com>

## Which problem is this PR solving?
- Remove the `smoke` E2E test suite since this test is covered by the `examples::simplest` test.

### CODE REVIEW

**Review by frzifus — APPROVED** [review]

**Comment by iblancasa** [comment]
@frzifus thanks for your review!

**Comment by iblancasa** [comment]
@rubenvp8510 can you take a look to this PR? Thanks :)

File: Dockerfile.asserts [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
-COPY go.mod /go/src/github.com/jaegertracing/jaeger-operator/go.mod
-COPY go.sum /go/src/github.com/jaegertracing/jaeger-operator/go.sum
+COPY go.mod go.mod
+COPY go.sum go.sum
 RUN go mod download
 
-COPY . /go/src/github.com/jaegertracing/jaeger-operator/
+COPY tests tests
+
+ENV GO111MODULE=on
```
Comment (frzifus) [inline Dockerfile.asserts:15]: I think thats a default since 1.16

Comment (iblancasa) [reply Dockerfile.asserts]: Sorry, I merged something should not be here. Thanks!

**Review by frzifus — DISMISSED** [review]

**Review by kevinearls — DISMISSED** [review]
LGTM

**Review by frzifus — APPROVED** [review]

**Review by rubenvp8510 — APPROVED** [review]

**Pull request merged**